### PR TITLE
PostgreSQL Alerts

### PR DESCRIPTION
**Pull request by himanshu219**
[List of  PostgreSQL Alerts](https://docs.google.com/spreadsheets/d/1dAxte2TFc7JDVZeukSJIXHSo2pOcfoalcYTeb4XzkYw/edit#gid=0)

### CODE REVIEW

File: monitor_packages/postgresql/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    {
+      connection_type       = "Email",
+      recipients            = ["abc@example.com"],
+      subject               = "Monitor Alert: {{TriggerType}} on {{Name}}",
+      time_zone             = "PST",
+      message_body          = "Triggered {{TriggerType}} Alert on {{Name}}: {{QueryURL}}",
+      run_for_trigger_types = ["Critical", "ResolvedCritical"]
+    }
+  ]
+}
```
Comment (arunpatyal) [inline monitor_packages/postgresql/README.md:70]: extra bracket

File: monitor_packages/postgresql/README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# sumologic-postgresql-monitors
```
Comment (arunpatyal) [inline monitor_packages/postgresql/README.md:1]: This readme will move to Sumo logic help, we just need to point to Sumo Help, check redis or nginx.

File: monitor_packages/postgresql/variables.tf [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    condition = contains([
+      "US1",
+    "us1","US2","us2","AU","au","CA","ca","DE","de","EU","eu","FED","fed","JP","jp","IN","in"], var.environment)
+    error_message = "Argument \"environment\" must be one of \"us1\",\"us2\",\"au\",\"ca\",\"de\",\"eu\",\"fed\",\"jp\",\"in\"."
+  }
+}
+variable "folder" {
+  type = string
+  description = "Folder where monitors will be created."
+  default = "Kubernetes"
```
Comment (arunpatyal) [inline monitor_packages/postgresql/variables.tf:30]: Change default to Postgres

File: monitor_packages/postgresql/postgresql.tf [hunk 4]
```diff
@@ -0,0 +1,377 @@
+# Sumo Logic Redis Metric Monitors
```
Comment (arunpatyal) [inline monitor_packages/postgresql/postgresql.tf:1]: Reference to Redis :)

**Review by arunpatyal — CHANGES_REQUESTED** [review]

**Comment by arunpatyal** [comment]
You can also add a custom cluster filter like dashboards so as to restrict the alerts to specific clusters or is postgres_data_source supposed to do that?

File: monitor_packages/postgresql/postgresql.json [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+  "children": [
+    {
+      "name": "Postgresql- High Rate Deadlock",
+      "description": "This alert fires when we detect deadlocks in a Postgres instance",
+      "type": "MonitorsLibraryMonitorExport",
+      "monitorType": "Metrics",
+      "queries": [
+        {
+          "rowId": "A",
+          "query": "$$postgresql_data_source metric=postgresql_deadlocks db_system=postgresql db_cluster=* db=* host=* | quantize using sum| sum by  db_cluster, host, db | rate | eval(_value*60)"
```
Comment (arunpatyal) [inline monitor_packages/postgresql/postgresql.json:14]: As you have custom source category, remember to mention in the readme to find and replace.

File: monitor_packages/postgresql/README.md (hunk 1 above)
Comment (himanshu219) [reply monitor_packages/postgresql/README.md:70]: updated readme

File: monitor_packages/postgresql/README.md (hunk 2 above)
Comment (himanshu219) [reply monitor_packages/postgresql/README.md:1]: done

File: monitor_packages/postgresql/variables.tf (hunk 3 above)
Comment (himanshu219) [reply monitor_packages/postgresql/variables.tf:30]: done

File: monitor_packages/postgresql/postgresql.tf (hunk 4 above)
Comment (himanshu219) [reply monitor_packages/postgresql/postgresql.tf:1]: done

File: monitor_packages/postgresql/postgresql.json (hunk 5 above)
Comment (himanshu219) [reply monitor_packages/postgresql/postgresql.json:14]: done in docs

**Comment by himanshu219** [comment]
> You can also add a custom cluster filter like dashboards so as to restrict the alerts to specific clusters or is postgres_data_source supposed to do that? @himanshu219

yes that is supposed to do that similar to how we do in app. Since it is not specific to cluster filtering customer can also use things like environment=dev and db_cluster=postgres or environment=teamA-dev so I named it as source filter.

**Review by arunpatyal — APPROVED** [review]

**Pull request merged**